fesvr/memif.h

```diff
@@ -79,4 +80,15 @@ class memif_t
   chunked_memif_t* cmemif;
 };
 
+class incompat_xlen : public std::exception {
+  private:
+  std::string msg;
+  
+  public:
+  incompat_xlen(const std::string& _msg) : msg(_msg) {};
+  const char* what() {
```

Comment:
Having `what()` return a string that's actually a number obscures what the reported problem is. Why not have two public fields in the exception class:
```
  const unsigned expected_xlen;
  const unsigned actual_xlen;
```
Then the contents of the exception object become self-explanatory. No need for `what()` at all.